pkg/providers/gcp/cluster_vpc.go

```diff
@@ -0,0 +1,52 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp/gcpiface"
+	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	errorUtil "github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	utils "k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getClusterVpc(ctx context.Context, c client.Client, networkClient gcpiface.NetworksAPI, logger *logrus.Entry) (*computepb.Network, error) {
```

Comment:
We could have a separate test suite for this function. This way we would be able to test all possible paths in isolation, without having to add too much mocks to the code that depends on this function.

Reply:
Sure will add a test for this